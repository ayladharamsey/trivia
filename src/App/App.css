* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.app {
  background-image:url('https://media.giphy.com/media/MaThe6p8WAKbf9NDDM/giphy.gif');
  background-size: 100%;
  background-position: center;
  font-family: Roboto;
  height: 100vh;
  width: 100vw;
  text-align: center;
  display: flex;
  justify-content: center;
}


h1 {
  color: #66FCF1;
  font-size: 1.7em;
  font-weight: 300;
}

h3 {
  color:#FFFFFF;
  font-size: 1em;
  font-weight: 500; 
  margin-top: 18px;
  margin-bottom: 7px;

}

input {
  width: 200px;
  height: 30px;
  border-style: none;
  border-radius: 5px;
}

button {
  width: 100px;
  height: 30px;
  font-weight: 700;
  border: 2px solid #66FCF1;
  background-color: #0B0C10;
  color:#66FCF1;
  border-radius: 5px;
  display: block;
  margin-left:37%;
  margin-top: 30px;
}

button:hover {
  background-color: #66FCF1;
  color: #0B0C10;
}

button:disabled {
  background-color: #0B0C10;
  color:#66FCF1;
}

a {
  text-decoration: none;
}



