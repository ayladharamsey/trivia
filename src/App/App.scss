* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

@font-face {font-family: "STARWARS";
  src: url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.eot");
  src: url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.eot?#iefix") format("embedded-opentype"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.woff") format("woff"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.ttf") format("truetype"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.svg#STARWARS") format("svg");
}

.app {
  background-image:url('https://media.giphy.com/media/MaThe6p8WAKbf9NDDM/giphy.gif'); 
  background-size: 100%;
  background-position: center;
  font-family: Roboto;
  height: 100vh;  
}

h1 {
  font-family: 'STARWARS';
  color: #feda4a;
  font-size: 1.7em;
  font-weight:600;
}

h3 {
  color:#FFFFFF;
  font-size: .9em;
  font-weight: 700; 
  margin-top: 5px;
  margin-bottom: 7px;
}

input {
  width: 200px;
  height: 30px;
  border-style: none;
  border-radius: 5px;
}

button {
  width: 100px;
  height: 30px;
  font-weight: 700;
  border: 2px solid #66FCF1;
  background-color: #0B0C10;
  color:#66FCF1;
  border-radius: 5px;
}

button:hover {
  background-color: #66FCF1;
  color: #0B0C10;
}

button:disabled {
  background-color: #0B0C10;
  color:#66FCF1;
}

a {
  text-decoration: none;
}



