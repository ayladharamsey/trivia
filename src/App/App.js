import React, { Component } from 'react';
import { movieTitles, getCharacters, getPeople, getSpecies, getPlanets, getResidents } from '../apiCalls';
import Form from '../Form/Form';
import './App.scss';
import HeaderContainer from '../HeaderContainer/HeaderContainer'
import Container from '../Container/Container';
import { Switch, Route} from 'react-router-dom'
import { createBrowserHistory } from 'history';
import OpeningCrawl from '../OpeningCrawl/OpeningCrawl';
const history = createBrowserHistory();

class App extends Component {
  constructor() {
    super();
    this.state = {
      movieData: [],
      characterData: [],
      planetData: [],
      favoriteCards: [1, 2  ],
      name: '',
      quote: '',
      rating: '',
      errMsg: ''
    }
  }

  signOut = () => {
    this.setState({
      name: '',
      quote: '',
      rating: '',
    })
  }

  updateAppState = (formState) => {
    console.log('updateAppState', formState);
    this.setState({
      name: formState.name, 
      quote: formState.quote,
      rating: formState.rating 
    })
  }

  updateFavoriteCard = (favoriteStatus) => {
    if(!this.state.favoriteCards.includes(favoriteStatus)) {
      favoriteStatus[6] = !favoriteStatus[6]
      this.setState({ favoriteCards: [...this.state.favoriteCards, favoriteStatus]});
    } else {
      favoriteStatus[6] = !favoriteStatus[6];
      const removeCard = this.state.favoriteCards.filter(card => card !== favoriteStatus)
      this.setState({ favoriteCards: removeCard });
    } 
  }
  componentDidMount = () => {
    movieTitles('https://swapi.co/api/films')
    .then(data => this.setMovie(data))
    .catch(error => console.log('Holy bat smoke, something went wrong in App!'))
  
    getCharacters('https://swapi.co/api/people/')
    .then(data => getPeople(data.results))
    .then(data => getSpecies(data))
    .then(data => this.setCharacters(data))

    getPlanets('https://swapi.co/api/planets/')
    // .then(data => console.log('planet data is : ', data))
    .then(data => getResidents(data.results))
    .then(data => this.setPlanets(data))
  }

  setMovie = (movieData) => {
    movieData.sort((a, b) => {
      return a.episode_id - b.episode_id
    });
    this.setState({ movieData })
  }

  setCharacters = (characterData) => {
    this.setState({ characterData })
  }
  setPlanets = (planetData) => {
    this.setState({ planetData })
  }

  render() {
    const { movieData } = this.state;
    return (
      <main className="app">
          <Switch>
            <Route 
              path='/' 
              exact 
              render={() => <Form history={history} updateAppState={this.updateAppState}/>}
            />
            <Route
              path='/movies'  
              exact
              render={ () => 
                <>
                  <HeaderContainer 
                    name={this.state.name} 
                    quote={this.state.quote} 
                    rating={this.state.rating} 
                    signOut={this.signOut} 
                  />
                  <Container movieData={this.state.movieData}/>
                </>
              }
            />
            <Route
              path='/movies/:id/characters'  
              exact
              render={ () => 
                <>
                  <HeaderContainer 
                    name={this.state.name} 
                    quote={this.state.quote} 
                    rating={this.state.rating} 
                    signOut={this.signOut} 
                  />
                  <Container 
                    charactersData={this.state.charactersData} 
                    planetsData={this.state.planetsData}
                    favoriteCards={this.state.favoriteCards}
                  />
                </>
              }
            />
            <Route
              path='/movies/:id/characters/favorites'  
              exact
              render={ () => 
                <>
                  <HeaderContainer 
                    name={this.state.name} 
                    quote={this.state.quote} 
                    rating={this.state.rating} 
                    signOut={this.signOut} 
                  />
                  <Container 
                    charactersData={this.state.charactersData} 
                    planetsData={this.state.planetsData}
                    favoriteCards={this.state.favoriteCards}
                  />
                </>
              }
            />
            <Route
              exact path='/movies/:id'
              render={ ( {match} ) => { 
                const { id } = match.params;
                const numId = parseInt(id)
                return(
                  <OpeningCrawl 
                      title={movieData[numId - 1].title}
                      date={movieData[numId - 1].release_date}
                      episode={movieData[numId - 1].episode_id}
                      text={movieData[numId - 1].opening_crawl}
                      favoriteCards={this.state.favoriteCards}
                      characters={this.state.movieData.characters}
                    />
                 )
                }   
              }
            /> 
          </Switch>
      </main>
    )
  }
}

export default App;
