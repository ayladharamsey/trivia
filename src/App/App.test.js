import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App Functionality ', () => {

  beforeEach(() => {

  });

  it('should match the snapshot', () => {

  });

  it('should have an empty state initially', () => {

  });

  it('should be able to sign out of the application', () => {

  });

  it('should be able update the name, quote, and rating when the form is filled out', () => {

  });

  it('should be able to update state by favoriting a card', () => {

  });

  it('should be able to update state with  character data', () => {

  });

  it('should be able to update state with movie data ', () => {

  });

  it('should be able to update state with planet data ', () => {

  });

});
