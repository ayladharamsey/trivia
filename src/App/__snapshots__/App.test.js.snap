// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Functionality  should match the snapshot 1`] = `
<main
  className="app"
>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/movies"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/movies/:id/characters"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/movies/:id/characters/favorites"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/movies/:id"
      render={[Function]}
    />
  </Switch>
</main>
`;
